src/snowflake/snowpark/_internal/utils.py

```diff
@@ -675,6 +680,25 @@ def warning(self, text: str) -> None:
         self.count += 1
 
 
+# TODO: SNOW-1720855: Remove DummyLock and DummyThreadLocal after the rollout
+class DummyLock:
+    """This is a dummy lock that is used in place of threading.Rlock when multithreading is
+    disabled."""
+
+    def __enter__(self):
```

Comment:
This seems requiring the lock class to be used with WithStatement, otherwise it will throw an error. You might want to add a comment somewhere to notify the developer about this

Reply:
I'll add methods for `acquire` and `release`. I expect this to be removed soon after we roll out thread-safe session for everyone.

Reply:
so typically we will remove things after 1 or 2 quaters after rollout to be safe